mlflow/tensorflow/__init__.py

```diff
@@ -1346,3 +1384,67 @@ def _log_tensorflow_dataset(tensorflow_dataset, source, context, name=None, targ
         return
 
     mlflow.log_input(dataset, context)
+
+
+def load_checkpoint(model=None, run_id=None, epoch=None, global_step=None):
+    """
+    If you enable "checkpoint" in autologging, during Keras model
+    training execution, checkpointed models are logged as MLflow artifacts.
+    Using this API, you can load the checkpointed model.
+
+    If you want to load the latest checkpoint, set both `epoch` and `global_step` to None.
+    If "checkpoint_save_freq" is set to "epoch" in autologging,
+    you can set `epoch` param to the epoch of the checkpoint to load specific epoch checkpoint.
+    If "checkpoint_save_freq" is set to an integer in autologging,
+    you can set `global_step` param to the global step of the checkpoint to load specific
+    global step checkpoint.
+    `epoch` param and `global_step` can't be set together.
+
+    Args:
+        model: A Keras model, this argument is required
+            only when the saved checkpoint is "weight-only".
+        run_id: The id of the run which model is logged to. If not provided,
+            current active run is used.
+        epoch: The epoch of the checkpoint to be loaded, if you set
+            "checkpoint_save_freq" to "epoch".
+        global_step: The global step of the checkpoint to be loaded, if
+            you set "checkpoint_save_freq" to an integer.
+
+    Returns:
+        The instance of a Keras model restored from the specified checkpoint.
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        mlflow.tensorflow.autolog(checkpoint=True, checkpoint_save_best_only=False)
+
+        model = create_tf_keras_model()  # Create a Keras model
+        with mlflow.start_run() as run:
+            model.fit(data, label, epoch=10)
+
+        run_id = run.info.run_id
+
+        # load latest checkpoint model
+        latest_checkpoint_model = mlflow.tensorflow.load_checkpoint(run_id=run_id)
+
+        # load history checkpoint model logged in second epoch
+        checkpoint_model = mlflow.tensorflow.load_checkpoint(run_id=run_id, epoch=2)
+    """
+    import tensorflow as tf
+
+    with TempDir() as tmp_dir:
+        downloaded_checkpoint_filepath = download_checkpoint_artifact(
+            run_id=run_id, epoch=epoch, global_step=global_step, dst_path=tmp_dir.path()
+        )
+
+        if os.path.basename(downloaded_checkpoint_filepath).split(".")[-2] == "weights":
```

Comment:
Improved error handling for this case:

```
        artifact_name = os.path.basename(downloaded_checkpoint_filepath)
        artifact_name_splits = artifact_name.split(".")
        if len(artifact_name_splits) < 2:
            raise MlflowException(
                f"The model checkpoint artifact file name '{artifact_name}' is malformed."
            )
```

Reply:
if a str doesn't contain dot, `'xx'.split(".")` gets `['xx']`

Reply:
I'd use `endswith`.

Reply:
Personally I prefer `split` than `regex`, if we incautiously write a bad regex it might cause performance issue when matching a long string, `split` is safer to me.

Reply:
@WeichenXu123 https://github.com/mlflow/mlflow/pull/11250/